### PR TITLE
fix: hide stack trace on syntax errors

### PR DESCRIPTION
**Pull request by Skn0tt**
Closes https://github.com/netlify/pod-dev-foundations/issues/580.

### CODE REVIEW

File: node/bundler.test.ts [hunk 1]
```diff
@@ -102,6 +103,15 @@ test('Adds a custom error property to user errors during bundling', async () =>
...
   } catch (error) {
     expect(error).toBeInstanceOf(BundleError)
+    expect((error as BundleError).message).toMatchInlineSnapshot(`
+      "error: Uncaught (in promise) Error: The module's source code could not be parsed: Unexpected eof at file:///root/functions/func1.ts:1:27
+
+        export default async () => 
+                                  ~
+            const ret = new Error(getStringFromWasm0(arg0, arg1));
+                        ^
+          at <anonymous> (file:///Users/skn0tt/dev/netlify/edge-bundler/deno/vendor/deno.land/x/eszip@v0.40.0/eszip_wasm.generated.js:417:19)"
```
Comment (eduardoboucas) [inline node/bundler.test.ts:113]: Your local paths will be a problem when matching the snapshot.

**Review by eduardoboucas — APPROVED** [review]
Nice one! You have a test failure, but the code looks good to me.

File: node/bundler.test.ts [hunk 2]
```diff
@@ -86,8 +86,9 @@ test('Uses the vendored eszip module instead of fetching it from deno.land', asy
   await cleanup()
 })
 
-test('Adds a custom error property to user errors during bundling', async () => {
-  expect.assertions(2)
+test.only('Adds a custom error property to user errors during bundling', async () => {
```
Comment (eduardoboucas) [inline node/bundler.test.ts:89]:
```suggestion
test('Adds a custom error property to user errors during bundling', async () => {
```

File: deno/bundle.ts [hunk 3]
```diff
@@ -3,4 +3,14 @@ import { writeStage2 } from './lib/stage2.ts'
 const [payload] = Deno.args
 const { basePath, destPath, externals, functions, importMapData } = JSON.parse(payload)
 
-await writeStage2({ basePath, destPath, externals, functions, importMapData })
+try {
+  await writeStage2({ basePath, destPath, externals, functions, importMapData })
+} catch (error) {
+  if (error instanceof Error) {
+    if (error.message.includes("The module's source code could not be parsed")) {
```
Comment (eduardoboucas) [inline deno/bundle.ts:10]: Could we not have a single `if` here?

```js
if (error instanceof Error && error.message.includes("The module's source code could not be parsed")) {
  // (...)
}
```

Comment (Skn0tt) [reply deno/bundle.ts]: done in 10bbe187cd6b8b727ce0d9fedc2ecf2996350ad2

File: node/bundle_error.ts [hunk 4]
```diff
@@ -30,6 +32,11 @@ class BundleError extends Error {
  */
 const wrapBundleError = (input: unknown, options?: BundleErrorOptions) => {
   if (input instanceof Error) {
+    if (input.message.includes("The module's source code could not be parsed")) {
```
Comment (eduardoboucas) [inline node/bundle_error.ts:35]: Same here, if you don't mind.

Comment (Skn0tt) [reply node/bundle_error.ts:35]: I don't think that works - we want to wrap it in a `BundleError`, even if the message is something different

Comment (eduardoboucas) [reply node/bundle_error.ts:35]: Oops, sorry. Didn't look at this properly.

**Review by eduardoboucas — APPROVED** [review]

**Pull request merged**